main.go

```diff
@@ -57,18 +59,83 @@ type TopologyResponseBody struct {
 	Type string `json:"type"`
 }
 
+type UnansweredNodeBroadcast struct {
+	Message uint64
+	Dest    string
+	SentAt  time.Time
+}
+
 func main() {
 	var messages []uint64
 	var topology map[string][]string
+	var unansweredNodeBroadcasts []UnansweredNodeBroadcast
+	// Prevent race condition on shared unansweredNodeBroadcasts access
+	var unansweredNodeBroadcastsMutex sync.Mutex
+	// Prevent race conditions when writing messages
+	var messagesMutex sync.Mutex
+
+	unansweredNodeBroadcastsEventLoopRunning := true
+
+	// Stop the event loop on node teardown
+	defer func() { unansweredNodeBroadcastsEventLoopRunning = false }()
 
 	n := maelstrom.NewNode()
 
+	// This goroutine acts over messages that haven't been answered for >=250ms and resends them
+	go func() {
+		var wg sync.WaitGroup
+
+		for unansweredNodeBroadcastsEventLoopRunning {
+			for _, unansweredNodeBroadcast := range unansweredNodeBroadcasts {
+				if time.Now().UTC().Add(-250*time.Millisecond).UnixMicro() <= unansweredNodeBroadcast.SentAt.UTC().UnixMicro() {
+					continue
+				}
+
+				// If the message hasn't been replied to >250ms, re-send it
+
+				wg.Add(1)
+
+				go func(destination string, message uint64) {
+					defer wg.Done()
+
+					n.Send(destination, BroadcastRequestBody{
```

Comment:
Should we update the `SentAt` timestamp here? As in 100ms (sleep time) we may have not received a reply yet